training/creating_dataset.py

```diff
@@ -15,117 +13,106 @@
-        "system/abilities",
-        "system/attributes/ac",
-        "system/attributes/hp",
-    ],
-) -> pd.DataFrame:
-    """
-    Creates dataframe containing chosen characteristics, level (CR) and source book of monsters from chosen books
-    :param books: list of paths of books to load
-    :param characteristics: list of characteristics to load
-    :return: DataFrame with monsters (NPC) form chosen books and with chosen characteristics and their origin book
-    """
-    loading_methods = {
-        "system/abilities": move_values_level_up(value_name="mod"),
-        "system/attributes/ac": load_subcolumn_as_value("ac"),
-        "system/attributes/hp": load_subcolumn_as_value("hp"),
-    }
-    for i in range(len(books) - 1, -1, -1):
-        if not is_path_correct(books[i]):
-            books.pop(i)
-
-    bestiary = get_merged_bestiaries(books)
-    bestiary = bestiary[bestiary["type"] == "npc"]
-    df = _create_df_with_basic_values(bestiary)
-
-    for characteristic in characteristics:
-        if (loading_method := loading_methods.get(characteristic)) is not None:
-            part = loading_method(get_subcolumn(bestiary, characteristic))
-            df = pd.concat(
-                [
-                    df,
-                    part,
-                ],
-                axis=1,
-            )
-
-    return df
+CHARACTERISTICS_COLUMNS = {
+    # "abilities": "abilities",
+    "cha": "abilities.cha.mod",
+    "con": "abilities.con.mod",
+    "mod": "abilities.dex.mod",
+    "int": "abilities.int.mod",
+    "str": "abilities.str.mod",
+    "wis": "abilities.wis.mod",
+    "ac": "attributes.ac.value",
+    "hp": "attributes.hp.value",
+    "perception": "attributes.perception.value",
+    # "saves": "saves"
+    "fortitude": "saves.fortitude.value",
+    "reflex": "saves.reflex.value",
+    "will": "saves.will.value",
+    "focus": "resources.focus.value",
+    "level": "details.level.value",
+    "book": "details.source.value",
+}
+"""dictionary with characteristics names (keys) and "path" to real columns in dataframe loaded from file (values)"""
 
 
 def is_path_correct(path: str) -> bool:
     """
     Checks whether given path is correct: file exists and correct file extension - json or db
+
     :param path: path to book to load
-    :return: True if path is correct otherwise False
+    :return: True if path is correct
     """
     if not os.path.isfile(path):
-        print(f"Path {path} does not exist")
-        return False
+        raise ValueError(f"Path {path} does not exist")
+        # return False
     if not (path.endswith(".db") or path.endswith(".json")):
-        print(f"Expected db or json file, got {path}")
-        return False
+        raise ValueError(f"Expected db or json file, got {path}")
+        # return False
     return True
 
 
-def move_values_level_up(value_name: str) -> callable:
+def load_and_preprocess_data(
+    paths_to_books: list[str] = None,
+    characteristics: list[str] = None,
```

Comment:
Założyłbym, że to musi być podane. To jest mocno wewnętrzny kod, używany tylko przy tworzeniu modelu. Nie możemy zakładać, że sami się walniemy i nie podamy parametru

Reply:
ok
Po prostu na samym początku było założenie, że mają być jakieś wartości domyślne